app/src/main/java/com/android/streetworkapp/utils/ParkFilter.kt

```diff
@@ -0,0 +1,153 @@
+package com.android.streetworkapp.utils
+
+import android.util.Log
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import androidx.compose.runtime.toMutableStateList
+import com.android.streetworkapp.model.park.Park
+
+/**
+ * This class add filtering capabilities to a list of parks.
+ *
+ * @param filterSettings The filter settings.
+ */
+class ParkFilter(private val filterSettings: FilterSettings) {
+
+  /**
+   * Filters a park based on the set filter settings.
+   *
+   * @param park The park to filter.
+   * @return True if the park passes the filter, false otherwise.
+   */
+  fun filter(park: Park): Boolean {
+    // Log.d("ParkFilter", "Filtering park ${park.name}")
+
+    return filterRating(park) && filterEventsDensity(park)
+  }
+
+  /**
+   * Filters a park according to the required minimal rating (filter settings).
+   *
+   * @param park The park to filter.
+   * @return True if the park passes the filter, false otherwise.
+   */
+  private fun filterRating(park: Park): Boolean {
+    return park.rating >= filterSettings.minRating.value
+  }
+
+  /**
+   * Filters a park according to its event density (filter settings).
+   *
+   * @param park The park to filter.
+   * @return The filtered list of parks.
+   */
+  private fun filterEventsDensity(park: Park): Boolean {
+    return filterSettings.eventDensity.any { it.isIncluded(park.events.size) }
+  }
+}
+
+/**
+ * Represents the filter settings.
+ *
+ * @property minRating The minimum rating required.
+ * @property eventDensity The required event density.
+ */
+class FilterSettings {
+  val minRating: MutableState<Int> = mutableIntStateOf(DEFAULT_RATING)
+  val eventDensity: SnapshotStateList<EventDensity> = DEFAULT_EVENT_DENSITY.toMutableStateList()
+
```

Comment:
You can already access and modify filterSettings values. This can be seen in `ParkFilterSettings` or in the `Tests` :

First, you instantiate the settings : 
>val filterSettings = FilterSettings()

Then, use the setter to change the default values (It also supports giving only one filter argument)
>filterSettings.set(minRating = 4, eventDensity = listOf(EventDensity.LOW)

Finally, you can access them manually in such calls : 
>val minRating = filterSettings.minRating.value
val eventDensities = filterSettings.enventDensity.value

You can also modifiy them directly (albeit the setter is here to protect the inside values, for instance trying to set a minmum rating greater than 5 would not change the inside value) : 
>filterSettings.minRating.value = 2

Reply:
You are right, it's fine as it is now